cpp/src/arrow/python/common.cc

```diff
@@ -68,20 +68,16 @@ Status CheckPyError(StatusCode code) {
   if (PyErr_Occurred()) {
     PyObject *exc_type, *exc_value, *traceback;
     PyErr_Fetch(&exc_type, &exc_value, &traceback);
-    PyObjectStringify stringified(exc_value);
+    PyErr_NormalizeException(&exc_type, &exc_value, &traceback);
+    PyObject *exc_value_str = PyObject_Str(exc_value);
+    char *c_str = PyString_AsString(exc_value_str);
```

Comment:
Yeah, I didn't notice that was only a Python2 function, let me try it again with PyObjectStringify